src/App/Pages/Accounts/LoginPasswordlessViewModel.cs

```diff
@@ -63,8 +70,30 @@ public LoginPasswordlessDetails LoginRequest
             }
         }
 
+        public void StopRequestTimeUpdater() => _requestTimeCts?.Cancel();
+
+        private void StartRequestTimeUpdater()
+        {
+            _requestTimeCts?.Cancel();
+            _requestTimeCts = new CancellationTokenSource();
+            _requestTimeTask = new TimerTask(_logger, UpdateRequestTime, _requestTimeCts).RunPeriodic(TimeSpan.FromMinutes(5));
+        }
+
+        private async void UpdateRequestTime()
+        {
+            TriggerPropertyChanged(nameof(TimeOfRequestText));
+            if (DateTime.UtcNow > LoginRequest?.RequestDate.ToUniversalTime().AddMinutes(15))
+            {
+                StopRequestTimeUpdater();
+                await _platformUtilsService.ShowDialogAsync(AppResources.LoginRequestHasAlreadyExpired);
+                await Page.Navigation.PopModalAsync();
+                return;
+            }
+        }
```

Comment:
Given that this is `async void` if an exception is thrown inside, it'll crash the application so a try...catch should be added here or convert this method to be a `Task` and either add the possibility of passing a `Func<Task>` to the `TimerTask` or just do a `FireAndForget` of the task depending on the desired behavior.
Also the `return` is not necessary given that it's the end of the method

Reply:
Adding the possibility to pass a Task to the `TimerTask` makes more sense.